packages/solid/store/src/server.ts

```diff
@@ -91,27 +91,27 @@ type ReconcileOptions = {
 };
 
 // Diff method for setStore
-export function reconcile<T>(
-  value: T | Store<T>,
+export function reconcile<T extends U, U>(
+  value: T,
   options: ReconcileOptions = {}
-): (state: Store<T>) => void {
+): (state: U) => T {
   return state => {
-    if (!isWrappable(state)) return value;
+    if (!isWrappable(state) || !isWrappable(value)) return value;
```

Comment:
I added a wrappable check to `value` to match the client version of `reconcile`, and afaict it makes sense to return if `value` is not wrappable since the whole object will be replaced. Also `Object.keys` will fail below if `value` is nullable. @ryansolid need your input on this change, and the added return statement at the end of this function to make it match the client which always returns either `state` or `value`.

Reply:
Looks good. Historically I have missed making updates in both places when I should have.

Reply:
Thanks!